src/components/common/TmBalance.vue

```diff
@@ -299,7 +305,16 @@ export default {
     // only be ready to withdraw of the validator rewards are loaded and the user has rewards to withdraw
     // the validator rewards are needed to filter the top 5 validators to withdraw from
     readyToWithdraw() {
-      return this.overview.totalRewards > 0
+      if (this.overview.rewards && this.overview.rewards.length > 0) {
```

Comment:
Why so complicated? Can't u just check if there is any record in rewards?

Reply:
? This is to prevent errors from overview & rewards not being loaded yet.

I am not sure what you mean

Reply:
I mean the other lines in this function. Why do you need to calculate "allTotalRewards"? Any reward is enough to allow withdraw right?

Reply:
Because rewards are coming from the API divided by validator at the moment

Reply:
but you can withdraw if you have any validator with any amount of rewards, right?

Reply:
nope. That is what `readyToWithdraw` is precisely doing ^^

Reply:
replace 309-314 with a check if there are any records in rewards. no need for all this logic...

Reply:
Look at my commits. I started like that. But the problem is that rewards are spread across validators... Do you see the issue? You'd need to wait for rewards to be above 0.001 on a single validator.

You can also do filter rewards with positive amounts, I agree. But users could withdraw then "invisible" rewards they cannot see in the UI, and it doesn't make sense since you are paying more fees than rewards received.

Reply:
I see. Good thinking. Can you comment those thoughts?